state/spacesdiscovery_test.go

```diff
@@ -429,6 +429,24 @@ func (s *SpacesDiscoverySuite) TestReloadSubnetsWithFAN(c *gc.C) {
 	checkSubnetsEqual(c, subnets, twoSubnetsAfterFAN)
 }
 
+func (s *SpacesDiscoverySuite) TestReloadSubnetsIgnoredWithFAN(c *gc.C) {
+	s.environ = networkedEnviron{
+		stub:           &testing.Stub{},
+		spaceDiscovery: false,
+		subnets:        twoSubnetsAndIgnored,
+	}
+	s.usedEnviron = &s.environ
+
+	s.IAASModel.UpdateModelConfig(map[string]interface{}{"fan-config": "fe80:dead:beef::/48=fe80:dead:beef::/24"}, nil)
```

Comment:
We could add a comment here that this should ultimately be changed once we update 'fan-config' to reject IPv6 sections.
Arguably we should already reject any fan-config that maps a region onto itself, regardless of whether it is IPv6 or IPv4.

Anyway, just a comment here "this configuration may be considered invalid in the future", so people who might make that change can fix this without worrying that it was something we really wanted to support.